File: templates/xcode/Hub Framework/Content operation.xctemplate/___FILEBASENAME___.m
#import "___FILEBASENAME___.h"

NS_ASSUME_NONNULL_BEGIN

@implementation ___FILEBASENAMEASIDENTIFIER___

@synthesize delegate = _delegate;

- (void)performForViewURI:(NSURL *)viewURI
              featureInfo:(id<HUBFeatureInfo>)featureInfo
        connectivityState:(HUBConnectivityState)connectivityState
         viewModelBuilder:(id<HUBViewModelBuilder>)viewModelBuilder
            previousError:(nullable NSError *)previousError
{
    // Perform the content operation, and call the delegate once done
}

@end

NS_ASSUME_NONNULL_END

File: templates/xcode/Hub Framework/Cont operation + initial.xctemplate/Swift/___FILEBASENAME___.swift
import Foundation
import HubFramework

class ___FILEBASENAMEASIDENTIFIER___: NSObject, HUBContentOperationWithInitialContent {
    weak var delegate: HUBContentOperationDelegate?

    func addInitialContent(viewURI: URL, viewModelBuilder: HUBViewModelBuilder) {
        // Optionally add any initial, "skeleton" view content to display while loading
    }

    func perform(forViewURI viewURI: URL, featureInfo: HUBFeatureInfo, connectivityState: HUBConnectivityState, viewModelBuilder: HUBViewModelBuilder, previousError: Error?) {
        // Perform the content operation, and call the delegate once done
    }
}
